### PR TITLE
Can't directly push, so here you go

### PR DESCRIPTION
**Pull request by theotherjimmy**
### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

---
This change is [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/79)

### CODE REVIEW

**Comment by mikisch81** [comment]
/morph build
Failure not relevant to this PR, related to spm_hal

File: targets/targets.json [hunk 1]
```diff
@@ -35,45 +35,53 @@
     "PSA_Target": {
         "public": false,
         "config": {
-            "secure-flash-start": {
+            "secure-rom-start": {
                 "help": "Starting address of Secure flash memory",
                 "value": null,
-                "macro_name": "PSA_SECURE_FLASH_START"
+                "macro_name": "PSA_SECURE_FLASH_START",
```
Comment (alzix) [inline targets/targets.json]: perhaps also should be renamed to `PSA_SECURE_ROM_START`

**Comment by alzix** [comment]
___
*[targets/targets.json, line 48 at r1](https://reviewable.io/reviews/kfnta/mbed-os/79#-LR8mS8j9S013GmSpCIC:-LR8mS8j9S013GmSpCID:bzgzvvq) ([raw file](https://github.com/kfnta/mbed-os/blob/28bf68036399d335131bdcbff38008661b2a8087/targets/targets.json#L48)):*
> ```JSON
>                 "value": null,
>                 "macro_name": "PSA_SECURE_FLASH_SIZE",
>                 "conflicts": ["target.mbed_rom_size"]
> ```

it seems to me that the `conflicts` property is redundant.
when you take a step back and look at them all - you can actually see a pattern :)
`secure-rom-size` and `non-secure-rom-size` will always conflict with `mbed_rom_size`
etc.

File: targets/targets.json (hunk 1 above)
Comment (theotherjimmy) [reply targets/targets.json]: That's not my change. I added a `,`.

**Comment by theotherjimmy** [comment]
> it seems to me that the conflicts property is redundant.

@alzix a while back
>having hard-coded list of artifacts may not provide us required flexibility in the future.
>i suggest sticking with our current data driven design, where all the data is encoded in config files and pure logic is found is scripts

So, data in the json, logic in the scripts :)

**Comment by mikisch81** [comment]
> So, data in the json, logic in the scripts :)

@alzix looks like Jimmy has plans for this new `conflict` member in other configuration properties.. 😸

File: targets/targets.json (hunk 1 above)
Comment (mikisch81) [reply targets/targets.json]: @alzix I agree, we can do it ourselves today (cause Jimmy will only be available afternoon).

Comment (mikisch81) [reply targets/targets.json]: Handled in #79

**Comment by mikisch81** [comment]
@alzix If its fine by you approve and merge

**Comment by theotherjimmy** [comment]
Thank guys!

**Pull request merged**